sources/packages/backend/apps/api/src/route-controllers/student-loan-balance/_tests_/e2e/student-loan-balance.aest.controller.getStudentLoanBalance.e2e-spec.ts

```diff
@@ -0,0 +1,149 @@
+import { HttpStatus, INestApplication } from "@nestjs/common";
+import * as request from "supertest";
+import {
+  AESTGroups,
+  BEARER_AUTH_TYPE,
+  createTestingAppModule,
+  getAESTToken,
+} from "../../../../testHelpers";
+import { MAXIMUM_ESSENTIAL_LOAN_BALANCE_RECORDS } from "../../../../utilities";
+
+import {
+  E2EDataSources,
+  createE2EDataSources,
+  saveFakeStudent,
+} from "@sims/test-utils";
+import { createFakeStudentLoanBalance } from "@sims/test-utils/factories/student-loan-balance";
+import { addToDateOnlyString } from "@sims/utilities";
+
+describe("StudentLoanBalanceAESTController(e2e)-getStudentLoanBalance", () => {
+  let app: INestApplication;
+  let db: E2EDataSources;
+
+  beforeAll(async () => {
+    const { nestApplication, dataSource } = await createTestingAppModule();
+    app = nestApplication;
+    db = createE2EDataSources(dataSource);
+  });
+
+  it(
+    "Should get all the student loan balance records," +
+      ` when a student has less than ${MAXIMUM_ESSENTIAL_LOAN_BALANCE_RECORDS} loan balance records.`,
+    async () => {
+      // Arrange
+      const student = await saveFakeStudent(db.dataSource);
+      // Create 4 loan balance records for the student.
+      const studentBalanceRecords = createStudentBalanceRecords(4, 100);
+      const studentLoanBalance = studentBalanceRecords.map((record) => {
+        return createFakeStudentLoanBalance(
+          { student },
+          {
+            initialValues: {
+              cslBalance: record.cslBalance,
+              balanceDate: record.balanceDate,
+            },
+          },
+        );
+      });
+      await db.studentLoanBalance.save(studentLoanBalance);
+      const endpoint = `/aest/student-loan-balance/student/${student.id}`;
+      const token = await getAESTToken(AESTGroups.BusinessAdministrators);
+
+      // Act/Assert
+      await request(app.getHttpServer())
+        .get(endpoint)
+        .auth(token, BEARER_AUTH_TYPE)
+        .expect(HttpStatus.OK)
+        .expect((response) => {
+          expect(response.body.loanBalanceDetails).toStrictEqual(
+            studentBalanceRecords,
+          );
+        });
+    },
+  );
+
+  it(
+    `Should get the student loan balance up to recent ${MAXIMUM_ESSENTIAL_LOAN_BALANCE_RECORDS}` +
+      ` records, when a student has more than ${MAXIMUM_ESSENTIAL_LOAN_BALANCE_RECORDS} loan balance records.`,
+    async () => {
+      // Arrange
+      const student = await saveFakeStudent(db.dataSource);
+      // Create loan balance records more than maximum essential records in recent  for the student.
+      const studentBalanceRecords = createStudentBalanceRecords(
+        MAXIMUM_ESSENTIAL_LOAN_BALANCE_RECORDS + 2,
```

Comment:
Minor, `+ 2` works but `+ 1` would also work in the same way inserting one less record.